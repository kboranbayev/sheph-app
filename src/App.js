import React, { Component } from "react";
import "./App.css";
import { BrowserRouter } from "react-router-dom";
import Header from "./components/static/header.js";
import Footer from "./components/static/footer.js";
import Router from "./Router";

class App extends React.Component {
  state = {
    checked: localStorage.getItem("theme") === "dark",
    theme: localStorage.getItem("theme")
  };

  componentDidMount() {
    document
      .getElementsByTagName("HTML")[0]
      .setAttribute("data-theme", localStorage.getItem("theme"));
  }

  toggleThemeChange = () => {
    const { checked } = this.state;
    if (checked === false) {
      localStorage.setItem("theme", "dark");
      document
        .getElementsByTagName("HTML")[0]
        .setAttribute("data-theme", localStorage.getItem("theme"));
      this.setState({
        checked: true
      });
    } else {
      localStorage.setItem("theme", "light");
      document
        .getElementsByTagName("HTML")[0]
        .setAttribute("data-theme", localStorage.getItem("theme"));
      this.setState({
        checked: false
      });
    }
  };

  render() {
    return (
      <BrowserRouter>
        <div className="App-theme">
          <Header backColor="#0062b3" />
            <div className="main-content-container container-fluid px-4">
              <Router /><br/>
              <label className="switch">
              <input
                type="checkbox"
                // checked={this.state.checked}
                defaultChecked={this.state.checked}
                onChange={() => this.toggleThemeChange()}
              />
              <span className="slider round"/>
            </label>
              <p className="text-monospace">Switch Mode</p>
          </div>
          <Footer />
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
