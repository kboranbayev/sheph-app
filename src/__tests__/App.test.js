import React from "react";
import ReactDOM from "react-dom";
import { act as domAct } from "react-dom/test-utils";
import { act as testAct, create } from "react-test-renderer";
import App from "../App";
import Main from "../pages/main";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

let root;
domAct(() => {
  testAct(() => {
    root = create(<Main />);
  });
});

expect(root).toMatchSnapshot();
