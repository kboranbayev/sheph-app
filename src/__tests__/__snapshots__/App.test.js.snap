// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div>
    <nav
      brand="sheph-w-bg.png"
      className="navbar navbar-expand-md navbar-light"
      style={
        Object {
          "backgroundColor": "#0062b3",
        }
      }
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "white",
            "fontWeight": "900",
          }
        }
      >
        <img
          alt="logo"
          height="50"
          src="sheph-w-bg.png"
          width="50"
        />
         
        Sheph
      </a>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded="false"
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <form
            action="/search-results"
            className="form-inline"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="searchBar"
                name="search"
                placeholder="Search a name..."
                required={true}
                type="search"
              />
            </div>
            <button
              aria-label={null}
              className="btn btn-success"
              onClick={[Function]}
            >
              Search
            </button>
          </form>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Browse
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/about"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                }
              }
            >
              About
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/submit"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Submit
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div
    className="main-content-container container-fluid px-4"
  >
    <div
      className="fluid container"
    >
      <div
        className="mb-5"
      >
        <h1
          className="page_title mt-5"
        >
          Active Posts
        </h1>
        <h7>
          ***Posts will be automatically removed after 6 days from date created***
        </h7>
        <div
          className="row"
        />
        <div />
      </div>
    </div>
  </div>
  <footer
    className="footer text-center"
  >
    <div
      className="container"
    >
      <div
        className="footer-bottom"
      >
        <p
          className="text-xs-center"
          style={
            Object {
              "color": "grey",
            }
          }
        >
          Â©
          2019
           Sheph - All Rights Reserved | Contact us: 
          <a
            href="mailto:sheph@gmail.com"
            style={
              Object {
                "color": "#55ACEE",
              }
            }
            target="_top"
          >
            Sheph@gmail.com
          </a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
